vita3k/app/src/app_init.cpp

```diff
@@ -118,44 +118,40 @@ void init_paths(Root &root_paths) {
 #if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
     // XDG Data Dirs.
     auto env_home = getenv("HOME");
-    auto env_data_dirs = getenv("XDG_DATA_DIRS");
-    auto env_data_home = getenv("XDG_DATA_HOME");
-    auto env_cache_home = getenv("XDG_CACHE_HOME");
-    auto env_config_home = getenv("XDG_CONFIG_HOME");
+    auto XDG_DATA_DIRS = getenv("XDG_DATA_DIRS");
+    auto XDG_DATA_HOME = getenv("XDG_DATA_HOME");
+    auto XDG_CACHE_HOME = getenv("XDG_CACHE_HOME");
+    auto XDG_CONFIG_HOME = getenv("XDG_CONFIG_HOME");
 
-    if (env_data_home != NULL)
-        root_paths.set_pref_path(fs::path(env_data_home) / app_name / app_name / dir_sep);
+    if (XDG_DATA_HOME != NULL)
+        root_paths.set_pref_path(fs::path(XDG_DATA_HOME) / app_name / app_name / dir_sep);
 
-    if (env_config_home != NULL)
-        root_paths.set_config_path(fs::path(env_config_home) / app_name / dir_sep);
+    if (XDG_CONFIG_HOME != NULL)
+        root_paths.set_config_path(fs::path(XDG_CONFIG_HOME) / app_name / dir_sep);
     else if (env_home != NULL)
         root_paths.set_config_path(fs::path(env_home) / ".config" / app_name / dir_sep);
 
-    if (env_cache_home != NULL) {
-        root_paths.set_cache_path(fs::path(env_cache_home) / app_name / dir_sep);
-        root_paths.set_log_path(fs::path(env_cache_home) / app_name / dir_sep);
+    if (XDG_CACHE_HOME != NULL) {
+        root_paths.set_cache_path(fs::path(XDG_CACHE_HOME) / app_name / dir_sep);
+        root_paths.set_log_path(fs::path(XDG_CACHE_HOME) / app_name / dir_sep);
     } else if (env_home != NULL) {
         root_paths.set_cache_path(fs::path(env_home) / ".cache" / app_name / dir_sep);
         root_paths.set_log_path(fs::path(env_home) / ".cache" / app_name / dir_sep);
     }
 
-    // Don't assume that base_path is portable.
-    if (fs::exists(root_paths.get_base_path() / "data") && fs::exists(root_paths.get_base_path() / "lang") && fs::exists(root_paths.get_base_path() / "shaders-builtin"))
```

Comment:
The reason that this is here, is to allow the normal release archives to work without any effort from the end-user.

If it's removed, when someone attempts to run Vita3K from the release archive, Vita3K will probably crash due to missing assets. Unless the end user just so happens to have a valid shared_path w/ needed assets in .local/share. The new code assumes this exception is the rule, which may not be the case.

Really there should probably be some validate_shared_path() function that actually checks that a given path _actually_ has all of the needed assets, (Vita3K tends to crash in a very user-unfriendly way when this is not the case), but this was the bare minimum needed to keep the normal release archives working while also allowing the new XDG support to work at the same time.